### PR TITLE
Fix lint result being ingored for `include/dali`. Fix linter errors in `include/dali`.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

### CODE REVIEW

**Comment by mzient** [comment]
!build

File: cmake/lint.cmake [hunk 1]
```diff
@@ -51,6 +51,12 @@ execute_process(
   OUTPUT_QUIET
 )
 
+if(LINT_RESULT)
+    message(FATAL_ERROR "Lint failed: ${LINT_ERROR}")
+else()
+    message(STATUS "Lint OK (headers)")
+endif()
+
```
Comment (mzient) [inline cmake/lint.cmake]: ^^^ that was the missing part.

File: include/dali/core/cuda_error.h [hunk 2]
```diff
@@ -12,12 +12,13 @@
...
 
-#ifndef DALI_CORE_CUDA_ERROR_H
-#define DALI_CORE_CUDA_ERROR_H
+#ifndef DALI_CORE_CUDA_ERROR_H_
+#define DALI_CORE_CUDA_ERROR_H_
 
 #include <cuda_runtime_api.h>
 #include <cstring>
 #include <cstdio>
+#include <string>
```
Comment (klecki) [inline include/dali/core/cuda_error.h]: Could you add a new line to separate include files groups (system and project includes)? It allows to sort them using clang-format.

File: include/dali/core/small_vector.h [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 #include <cuda_runtime.h>
 #include <utility>
+#include <memory>
```
Comment (klecki) [inline include/dali/core/small_vector.h]: Also newline here.

**Pull request merged**